doc-examples/src/test/scala/akka/stream/alpakka/eip/scaladsl/PassThroughExamples.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.eip.scaladsl
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.kafka.ConsumerMessage.CommittableOffsetBatch
+import akka.kafka.{ConsumerMessage, ConsumerSettings, Subscriptions}
+import akka.kafka.scaladsl.Consumer
+import akka.kafka.scaladsl.Consumer.DrainingControl
+import akka.stream.{ActorMaterializer, FlowShape, Graph}
+import akka.stream.scaladsl._
+import org.apache.kafka.common.serialization.{ByteArrayDeserializer, StringDeserializer}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpec}
+
+import scala.concurrent.duration._
+
+class PassThroughExamples extends WordSpec with BeforeAndAfterAll with Matchers with ScalaFutures {
+
+  implicit val system = ActorSystem("Test")
+  implicit val mat = ActorMaterializer()
+
+  "PassThroughFlow" should {
+    " transform message but the original message is maintained " in {
+
+      //#PassThroughSimple
+      //Sample Source
+      val source = Source(List(1, 2, 3))
+
+      // Int to Unit
+      val printerFlow =
+        Flow[Int]
+          .map(println)
+
+      // Int to Int
+      val plus10Flow =
+        Flow[Int]
+          .map(_ + 10)
+
+      val ret = source
+        .via(PassThroughFlow(printerFlow))
+        .via(plus10Flow)
+        .runWith(Sink.seq)
+
+      //Verify results
+
+      ret.futureValue should be(Vector(11, 12, 13))
+      //#PassThroughSimple
+
+    }
+  }
+
+  override protected def afterAll(): Unit = system.terminate()
+}
+
+//#PassThrough
+object PassThroughFlow {
+  def apply[A, T](processingFlow: Flow[A, T, NotUsed]): Graph[FlowShape[A, A], NotUsed] =
+    Flow.fromGraph(GraphDSL.create() { implicit builder =>
+      {
+        import GraphDSL.Implicits._
+
+        val tuplerizer = builder.add(Flow[A].map(x => (x, x)))
+        val detuplerizer = builder.add(Flow[(T, A)].map(_._2))
+
+        val unzip = builder.add(Unzip[A, A]())
+        val zip = builder.add(Zip[T, A]())
+
+        val passThroughFlow = Flow[A].map(identity)
```

Comment:
You should remove `passThroughFlow` and connect the `merge.out(1)` directly with the `zip.in1`.
If you want to keep the Ascii-art, you may surround it with `// format:off` and `on`.

Reply:
Sorry, it is not clear to me how should I use `Merge` and `Broadcast` ... reading the documentation it is not clear how to use it in my use case. Can you provide an example. Why they are preferred instead of `Zip`/`Unzip`?